pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -0,0 +1,455 @@
+	params := []any{}
+
+	// 2- Filtering: WHERE clauses (from lo.Filters)
+	whereClauses := []string{}
+	for _, orFilters := range lo.Filters {
+		orClause, orParams := buildORClause(orFilters)
+		whereClauses = append(whereClauses, orClause)
+		params = append(params, orParams...)
+	}
+
+	// WHERE clauses (from namespace)
+	if namespace != "" && namespace != "*" {
+		whereClauses = append(whereClauses, fmt.Sprintf(`f."metadata.namespace" = ?`))
+		params = append(params, namespace)
+	}
+
+	// WHERE clauses (from partitions and their corresponding parameters)
+	partitionClauses := []string{}
+	for _, partition := range partitions {
+		if partition.Passthrough {
+			// nothing to do, no extra filtering to apply by definition
+		} else {
+			singlePartitionClauses := []string{}
+
+			// filter by namespace
+			if partition.Namespace != "" && partition.Namespace != "*" {
+				singlePartitionClauses = append(singlePartitionClauses, fmt.Sprintf(`f."metadata.namespace" = ?`))
+				params = append(params, partition.Namespace)
+			}
+
+			// optionally filter by names
+			if !partition.All {
+				names := partition.Names
+
+				if len(names) == 0 {
+					// degenerate case, there will be no results
+					singlePartitionClauses = append(singlePartitionClauses, "FALSE")
+				} else {
+					singlePartitionClauses = append(singlePartitionClauses, fmt.Sprintf(`f."metadata.name" IN (?%s)`, strings.Repeat(", ?", len(partition.Names)-1)))
+					for name := range partition.Names {
+						params = append(params, name)
+					}
+				}
+			}
+
+			if len(singlePartitionClauses) > 0 {
+				partitionClauses = append(partitionClauses, strings.Join(singlePartitionClauses, " AND "))
+			}
+		}
+	}
+	if len(partitions) == 0 {
+		// degenerate case, there will be no results
+		whereClauses = append(whereClauses, "FALSE")
+	}
+	if len(partitionClauses) == 1 {
+		whereClauses = append(whereClauses, partitionClauses[0])
+	}
+	if len(partitionClauses) > 1 {
+		whereClauses = append(whereClauses, "(\n      ("+strings.Join(partitionClauses, ") OR\n      (")+")\n)")
+	}
+
+	if len(whereClauses) > 0 {
+		stmt += "\n  WHERE\n    "
+		for index, clause := range whereClauses {
+			stmt += fmt.Sprintf("(%s)", clause)
+			if index == len(whereClauses)-1 {
+				break
+			}
+			stmt += " AND\n    "
+		}
+		// stmt += strings.Join(whereClauses, " AND\n    ")
+	}
+
+	// 2- Sorting: ORDER BY clauses (from lo.Sort)
+	orderByClauses := []string{}
+	if len(lo.Sort.PrimaryField) > 0 {
+		columnName := toColumnName(lo.Sort.PrimaryField)
+		direction := "ASC"
+		if lo.Sort.PrimaryOrder == DESC {
+			direction = "DESC"
+		}
+		orderByClauses = append(orderByClauses, fmt.Sprintf(`f."%s" %s`, columnName, direction))
+	}
+	if len(lo.Sort.SecondaryField) > 0 {
+		columnName := toColumnName(lo.Sort.SecondaryField)
+		direction := "ASC"
+		if lo.Sort.SecondaryOrder == DESC {
+			direction = "DESC"
+		}
+		orderByClauses = append(orderByClauses, fmt.Sprintf(`f."%s" %s`, columnName, direction))
```

Comment:
The column name seems to be end-user provided input. This seems like a case for injection to me. Can you give this a look and let me know what you think?

Reply:
Same as https://github.com/rancher/lasso/pull/65#discussion_r1612435288.